Source/WebCore/css/parser/CSSPropertyParserConsumer+Align.cpp

```diff
@@ -75,14 +75,22 @@ static bool isContentPositionOrLeftOrRightKeyword(CSSValueID id)
     return isContentPositionKeyword(id) || isLeftOrRightKeyword(id);
 }
 
-static bool isSelfPositionKeyword(CSSValueID id)
-{
-    return identMatches<CSSValueStart, CSSValueEnd, CSSValueCenter, CSSValueSelfStart, CSSValueSelfEnd, CSSValueFlexStart, CSSValueFlexEnd, CSSValueAnchorCenter>(id);
-}
+enum class AllowedSelfPositionKeywords {
+    LeftRight    = 1 << 0,
+    AnchorCenter = 1 << 1
+};
```

Comment:
I think this might be a bit over engineered, can we just pass a bool for anchor center allowed?

Reply:
The code right now passes a function (isSelfPositionKeyword and isSelfPositionOrLeftRightKeyword) to check if a keyword is a self position keyword or not. If we add a bool parameter to those functions, then the functions that call those functions must also accept the bool and pass it along. Also we'd have to make 4 check functions for left/right and anchor-center optional. So I guess might as well use OptionSet.

Reply:
The naming could be better though, maybe OptionalSelfPositionKeywords

Reply:
It's honestly fine either way, I didn't originally read the patch too much in detail but both probably are about same amount of code. Feel free to keep as is.